rules/Composer/Processor/ComposerProcessor.php

```diff
@@ -32,75 +31,64 @@ final class ComposerProcessor
     private $composerJsonPrinter;
 
     /**
-     * @var Configuration
-     */
-    private $configuration;
-
-    /**
-     * @var ErrorAndDiffCollector
-     */
-    private $errorAndDiffCollector;
-
-    /**
-     * @var SmartFileSystem
+     * @var ComposerModifier
      */
-    private $smartFileSystem;
+    private $composerModifier;
 
     /**
-     * @var ComposerModifier
+     * @var Configuration
      */
-    private $composerModifier;
+    private $configuration;
 
     public function __construct(
         ComposerJsonFactory $composerJsonFactory,
         ComposerJsonPrinter $composerJsonPrinter,
         Configuration $configuration,
-        ErrorAndDiffCollector $errorAndDiffCollector,
-        SmartFileSystem $smartFileSystem,
         ComposerModifier $composerModifier
     ) {
         $this->composerJsonFactory = $composerJsonFactory;
         $this->composerJsonPrinter = $composerJsonPrinter;
         $this->configuration = $configuration;
-        $this->errorAndDiffCollector = $errorAndDiffCollector;
-        $this->smartFileSystem = $smartFileSystem;
         $this->composerModifier = $composerModifier;
     }
 
-    public function process(string $composerJsonFilePath): void
+    public function process(SmartFileInfo $smartFileInfo): ?string
     {
-        if (! $this->smartFileSystem->exists($composerJsonFilePath)) {
-            return;
-        }
-
         // to avoid modification of file
         if (! $this->composerModifier->enabled()) {
-            return;
+            return null;
         }
 
-        $smartFileInfo = new SmartFileInfo($composerJsonFilePath);
         $composerJson = $this->composerJsonFactory->createFromFileInfo($smartFileInfo);
-
         $oldComposerJson = clone $composerJson;
         $this->composerModifier->modify($composerJson);
 
         // nothing has changed
         if ($oldComposerJson->getJsonArray() === $composerJson->getJsonArray()) {
-            return;
+            return null;
         }
+        $newContent = $this->composerJsonPrinter->printToString($composerJson);
 
-        $this->addComposerJsonFileDiff($oldComposerJson, $composerJson, $smartFileInfo);
         $this->reportFileContentChange($composerJson, $smartFileInfo);
+
+        return $newContent;
+    }
+
+    public function canProcess(SmartFileInfo $smartFileInfo): bool
+    {
+        return $smartFileInfo->getRealPath() === getcwd() . '/composer.json';
     }
 
-    private function addComposerJsonFileDiff(
-        ComposerJson $oldComposerJson,
-        ComposerJson $newComposerJson,
-        SmartFileInfo $smartFileInfo
-    ): void {
-        $newContents = $this->composerJsonPrinter->printToString($newComposerJson);
-        $oldContents = $this->composerJsonPrinter->printToString($oldComposerJson);
-        $this->errorAndDiffCollector->addFileDiff($smartFileInfo, $newContents, $oldContents);
+    public function allowedFileExtensions(): array
```

Comment:
This might be better name as it's about supported extensions. What do you think?

```suggestion
    public function getSupportedFileExtensions(): array
```